### PR TITLE
gurobi: init at 8.0.1

### PR DESCRIPTION
**Pull request by jfrankenau**
###### Motivation for this change

See #32513

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @bbarker @shlevy

### CODE REVIEW

File: pkgs/applications/science/math/gurobi/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  src = with stdenv.lib; fetchurl {
+    url = "http://packages.gurobi.com/${versions.majorMinor version}/gurobi${version}_linux64.tar.gz";
+    sha256 = "0y3lb0mngnyn7ql4s2n8qxnr1d2xcjdpdhpdjdxc4sc8f2w2ih18";
+  };
+
+  sourceRoot = "gurobi${builtins.replaceStrings ["."] [""] version}/linux64";
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [ (python.withPackages (ps: [ ps.gurobipy ])) ];
```
Comment (shlevy) [inline pkgs/applications/science/math/gurobi/default.nix]: This should work by just having `python` and `pythonPackages.gurobipy` here

Comment (jfrankenau) [reply pkgs/applications/science/math/gurobi/default.nix]: Without `python.withPackages` the Gurobi interactive shell (which is a python script) is not able to find the `gurobipy` module.

Comment (shlevy) [reply pkgs/applications/science/math/gurobi/default.nix]: @jfrankenau Even when `python` itself is in the buildInputs? Interesting, OK.

Comment (shlevy) [reply pkgs/applications/science/math/gurobi/default.nix]: Not that I know of, no.

**Pull request merged**